Mathlib/NumberTheory/FLT/Basic.lean

```diff
@@ -117,3 +119,43 @@ lemma fermatLastTheoremFor_iff_int {n : ℕ} : FermatLastTheoremFor n ↔ Fermat
 
 lemma fermatLastTheoremFor_iff_rat {n : ℕ} : FermatLastTheoremFor n ↔ FermatLastTheoremWith ℚ n :=
   (fermatLastTheoremWith_nat_int_rat_tfae n).out 0 2
+
+open Finset in
+/-- To prove Fermat Last Theorem in any semiring that is a `NormalizedGCDMonoid` one can assume
+that the `gcd` of `{a, b, c}` is `1`. -/
+lemma fermatLastTheoremWith_of_FermatLastTheoremWith_coprime {n : ℕ} {R : Type*} [CommSemiring R]
+    [IsDomain R] [DecidableEq R] [NormalizedGCDMonoid R]
+    (hn : ∀ a b c : R, a ≠ 0 → b ≠ 0 → c ≠ 0 → ({a, b, c} : Finset R).gcd id = 1 →
+      a ^ n + b ^ n ≠ c ^ n) :
+    FermatLastTheoremWith R n := by
+  intro a b c ha hb hc habc
+  let s : Finset R := {a, b, c}; let d := s.gcd id
+  obtain ⟨A, hA⟩ : d ∣ a := gcd_dvd (by simp [s])
+  obtain ⟨B, hB⟩ : d ∣ b := gcd_dvd (by simp [s])
+  obtain ⟨C, hC⟩ : d ∣ c := gcd_dvd (by simp [s])
+  simp only [hA, hB, hC, mul_ne_zero_iff, mul_pow] at ha hb hc habc
+  rw [← mul_add, mul_right_inj' (pow_ne_zero n ha.1)] at habc
+  refine hn A B C ha.2 hb.2 hc.2 ?_ habc
+  rw [← Finset.normalize_gcd, normalize_eq_one]
+  obtain ⟨u, hu⟩ := normalize_associated d
+  refine ⟨u, mul_left_cancel₀ (mt normalize_eq_zero.mp ha.1) (hu.symm ▸ ?_)⟩
+  rw [← Finset.gcd_mul_left, gcd_eq_gcd_image, image_insert, image_insert, image_singleton,
+      id_eq, id_eq, id_eq, ← hA, ← hB, ← hC]
+
+/-- To prove Fermat Last Theorem for `ℤ` one can assume that `a`, `b` and `c` are coprime, in the
+sense that the gcd of `{a, b, c}` is `1`. -/
+lemma FermatLastTheoremWith_int_of_FermatLastTheoremWith_int_coprime {n : ℕ}
+    (hn : ∀ a b c : ℤ, a ≠ 0 → b ≠ 0 → c ≠ 0 → ({a, b, c} : Finset ℤ).gcd id = 1 →
+      a ^ n + b ^ n ≠ c ^ n) :
+    FermatLastTheoremWith ℤ n :=
+  fermatLastTheoremWith_of_FermatLastTheoremWith_coprime
+    (fun a b c ha hb hc H h ↦ hn a b c ha hb hc H h)
+
+/-- To prove Fermat Last Theorem one can assume that `a`, `b` and `c` are coprime, in the sense
+that the gcd of `{a, b, c}` is `1`. -/
+lemma FermatLastTheoremFor_of_FermatLastTheoremFor_coprime {n : ℕ}
+    (hn : ∀ a b c, a ≠ 0 → b ≠ 0 → c ≠ 0 → ({a, b, c} : Finset ℕ).gcd id = 1 →
+      a ^ n + b ^ n ≠ c ^ n) :
+    FermatLastTheoremFor n :=
+  fermatLastTheoremWith_of_FermatLastTheoremWith_coprime (R := ℕ)
+    (fun a b c ha hb hc H h ↦ hn a b c ha hb hc (by simpa using H) h)
```

Comment:
Is there any point keeping these anymore?
```suggestion
/-- To prove Fermat Last Theorem for `ℤ` one can assume that `a`, `b` and `c` are coprime, in the
sense that the gcd of `{a, b, c}` is `1`. -/
lemma FermatLastTheoremWith_int_of_FermatLastTheoremWith_int_coprime {n : ℕ}
    (hn : ∀ a b c : ℤ, a ≠ 0 → b ≠ 0 → c ≠ 0 → ({a, b, c} : Finset ℤ).gcd id = 1 →
      a ^ n + b ^ n ≠ c ^ n) :
    FermatLastTheoremWith ℤ n :=
  fermatLastTheoremWith_of_FermatLastTheoremWith_coprime hn

/-- To prove Fermat Last Theorem one can assume that `a`, `b` and `c` are coprime, in the sense
that the gcd of `{a, b, c}` is `1`. -/
lemma FermatLastTheoremFor_of_FermatLastTheoremFor_coprime {n : ℕ}
    (hn : ∀ a b c, a ≠ 0 → b ≠ 0 → c ≠ 0 → ({a, b, c} : Finset ℕ).gcd id = 1 →
      a ^ n + b ^ n ≠ c ^ n) :
    FermatLastTheoremFor n :=
  fermatLastTheoremWith_of_FermatLastTheoremWith_coprime hn
```

Reply:
Probably not, you're right. Thanks!